### DISCUSSION TITLE
Release planned w Java 23 support?

### DISCUSSION THREAD

**Discussion by kennytv** [async-profiler/async-profiler#1057] [Q&A]
Hey, I know Java 23 compatibility has already been fixed a long while ago, but it'd be nice to have that in a stable release to depend on given Java 23 has been GA for some time now. Is there a new release version of async-profiler planned anytime soon?

**Comment by apangin** [comment]
Yes, new release is being prepared.
In the meantime, you are welcome to use [Nightly releases](https://github.com/async-profiler/async-profiler/releases/tag/nightly).

**Reply by dkandalov** [reply]
Unfortunately, in some organisations it's not possible to use nightly builds or compile project from the source code :(

**Reply by mr-git** [reply]
specially when application is used in some regulated market - those require fixed-version artifacts and repeatable builds.

**Comment by kashike** [comment]
You said a release was being prepared [a month ago](https://github.com/async-profiler/async-profiler/issues/923#issuecomment-2408543947) - is there any way to expedite the release?

**Reply by apangin** [reply]
Right - over the past month, we have considerably improved async-profiler release process. The project is automatically built and tested on all supported platforms after each commit, and on success, a new release is published just in 5 minutes since the last merge. I don't think it can be expedited any further. All workflows and dockerfiles are available in the repository. The release process of async-profiler has never been as rapid and transparent as today.

**Reply by fmeum** [reply]
The auto-release workflow is very useful for end users, but it's not quite sufficient for other projects depending on async-profiler. For example, Bazel embeds async-profiler and needs a stable download URL with a stable hash for all artifacts. But the nightly releases keep getting updated with new versions, so filenames and hashes change too often to keep up.

A tagged release that's kept around indefinitely would still be highly welcome once in a while.

**Reply by mr-git** [reply]
tagged release per JDK release sounds like a good cadence

**Reply by amotl** [reply]
Hi. First things first: Thanks a stack for your work on this excellent package.

We are also looking forward to a general release, so that downstream projects can use it.
I think one of the most prominent items is that people are eager to use Grafana Pyroscope with applications that use Java 23.

- https://github.com/grafana/pyroscope-java/issues/165

**Reply by mr-git** [reply]
In few hours JDK 24 will get released and its `Final Release Candidate` has been available for 4 weeks already.

Is there a way for us to help with release for latest and greatest JDK?

**Reply by bmunkholm** [reply]
@apangin Any chance you can inform when a tagged release will be available?

**Reply by apangin** [reply]
Hopefully this week. All known blockers have been resolved.
